### PR TITLE
Fixed crash if player restarts mission in SP.

### PR DESCRIPTION
**Pull request by nukeulater**
Also fixed the onGameEngineChange due to bad merge.
Some cleanup.
Added missing Initialize for H2 final gametype.

### CODE REVIEW

File: xlive/H2Tweaks.cpp [hunk 1]
```diff
@@ -414,3 +414,13 @@ void H2Tweaks::FixRanksIcons() {
 	}
 }
 
+void applyCrashFix() {
+
+	//Crashfix
```
Comment (shayaantx) [inline xlive/H2Tweaks.cpp]: if you're gonna comment this out, why keep it?

File: xlive/H2MOD.cpp [hunk 2]
```diff
@@ -1273,7 +1229,7 @@ void H2MOD::Initialize()
 
 		PatchGameDetailsCheck();
 		//H2Tweaks::PatchPingMeterCheck();
-		*(bool*)((char*)h2mod->GetBase() + 0x422450) = 1; //allows for all live menus to be accessed
+		*(bool*)(h2mod->GetBase() + 0x422450) = 1; //allows for all live menus to be accessed
```
Comment (shayaantx) [inline xlive/H2MOD.cpp]: why you change this?

**Comment by shayaantx** [comment]
who tested this?

**Comment by nukeulater** [comment]
me and general.

File: xlive/H2MOD.cpp (hunk 2 above)
Comment (nukeulater) [reply xlive/H2MOD.cpp]: no need of type-casting it to char* then to dereference it as bool, also i'm the one who added that a while ago.

File: xlive/H2Tweaks.cpp (hunk 1 above)
Comment (nukeulater) [reply xlive/H2Tweaks.cpp]: will fix

File: xlive/H2MOD.cpp [hunk 3]
```diff
@@ -229,7 +231,7 @@ void H2MOD::logToDedicatedServerConsole(wchar_t* message) {
 }
 
 int __cdecl dediCommandHook(int a1, int a2, int a3) {
-	h2mod->logToDedicatedServerConsole(L"Dedicated command\n");
+	//h2mod->logToDedicatedServerConsole(L"Dedicated command\n");
```
Comment (shayaantx) [inline xlive/H2MOD.cpp]: why?

Comment (nukeulater) [reply xlive/H2MOD.cpp]: Would be nice to log that only if a custom command has been used.

File: xlive/H2ConsoleCommands.cpp [hunk 4]
```diff
@@ -465,6 +465,7 @@ void ConsoleCommands::handle_command(std::string command) {
 		}
 		else {
 			output(L"Unknown command.");
+			h2mod->logToDedicatedServerConsole(L"No such custom command.");
```
Comment (shayaantx) [inline xlive/H2ConsoleCommands.cpp]: i looked at code, output will output to console, u can just remove this

Comment (nukeulater) [reply xlive/H2ConsoleCommands.cpp]: alright, thanks

**Pull request merged**